gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -797,39 +790,34 @@ OGRErr OGRGeoPackageTableLayer::ReadTableDefinition()
                         " LIMIT 2"
                         ,m_pszTableName);
 
-                err = SQLQuery(poDb, pszSQL, &oResultGeomCols);
+                oResultGeomCols = SQLQuery(poDb, pszSQL);
                 sqlite3_free(pszSQL);
             }
 
             /* gpkg_geometry_columns query has to work */
             /* gpkg_geometry_columns.table_name is supposed to be unique */
-            if ( err != OGRERR_NONE || oResultGeomCols.nRowCount != 1 )
+            if ( !oResultGeomCols || oResultGeomCols->RowCount() != 1 )
             {
-                if ( err != OGRERR_NONE )
```

Comment:
Small behavior change: I've removed this if because it's not possible anymore to get the error message since SQLQuery returns null on error right now. But anyway, SQLQuery already calls `CPLError` with the error message, so I believe this was printing the error message twice